packages/rum/src/domain/record/observers.spec.ts

```diff
@@ -328,3 +333,64 @@ describe('initStyleSheetObserver', () => {
     })
   })
 })
+
+describe('initMouseInteractionObserver', () => {
+  let mouseInteractionCallbackSpy: jasmine.Spy<MouseInteractionCallBack>
+  let stopObserver: () => void
+  let sandbox: HTMLDivElement
+  let a: HTMLAnchorElement
+  let coordinatesComputed: boolean
+
+  beforeEach(() => {
+    if (isIE()) {
+      pending('IE not supported')
+    }
+    if (!window.visualViewport) {
+      pending('no visualViewport')
+    }
+
+    sandbox = document.createElement('div')
+    a = document.createElement('a')
+    a.setAttribute('tabindex', '0') // make the element focusable
+    sandbox.appendChild(a)
+    document.body.appendChild(sandbox)
+    a.focus()
+
+    serializeDocument(document, DEFAULT_CONFIGURATION, {
+      shadowRootsController: DEFAULT_SHADOW_ROOT_CONTROLLER,
+      status: SerializationContextStatus.INITIAL_FULL_SNAPSHOT,
+      elementsScrollPositions: createElementsScrollPositions(),
+    })
+
+    coordinatesComputed = false
+    Object.defineProperty(window.visualViewport, 'offsetTop', {
+      get() {
+        coordinatesComputed = true
+        return 0
+      },
+      configurable: true,
+    })
+
+    mouseInteractionCallbackSpy = jasmine.createSpy()
+    stopObserver = initMouseInteractionObserver(mouseInteractionCallbackSpy, DefaultPrivacyLevel.ALLOW)
+  })
+
+  afterEach(() => {
+    sandbox.remove()
+    delete (window.visualViewport as any).offsetTop
+    stopObserver()
+  })
+
+  it('should compute x/y coordinates for click record', () => {
+    a.click()
+    expect(mouseInteractionCallbackSpy).toHaveBeenCalled()
+    expect(coordinatesComputed).toBeTrue()
+  })
+
+  // related to safari issue, see RUMF-1450
+  it('should not compute x/y coordinates for blur record', () => {
+    a.blur()
+    expect(mouseInteractionCallbackSpy).toHaveBeenCalled()
```

Comment:
**suggestion(not-blocking):** use `toHaveBeenCalledWith` to check that x and y are present in the payload